### ISSUE TITLE
Creating a chained KSPromise via the then:error: method silently fails if the promise is cancelled

### ISSUE THREAD

**Issue by xtreme-amanda-kehm** [kseebaldt/deferred#53]
When attempting to chain to a promise via then:error: method, if the promise is cancelled nil is returned and the requester is potentially waiting on something that will never complete.

**Comment by tomquist** [comment]
I created a PR that fixes the issue: https://github.com/kseebaldt/deferred/pull/60
